plato/file_storage.py

```diff
@@ -0,0 +1,208 @@
+
+        static_files = os.listdir(static_path(base_tmp_path, template_id))
+        for static_file in static_files:
+            tmp_static_sys_path = Path(static_file_path(base_tmp_path, template_id, static_file))
+            with tmp_static_sys_path.open(mode='rb') as tmp_file:
+                self.save_file(tmp_file, static_file_path(template_dir, template_id, static_file))
+
+    @abstractmethod
+    def save_file(self, input_file: BinaryIO, path: str) -> None:
+        """
+        Write file into storage folder
+
+        Args:
+            input_file (BinaryIO): the input file
+            path (str): the storage path
+        """
+        raise NotImplementedError
+
+    def write_file_locally(self, input_file: BinaryIO, path: str):
+        """
+        Writes a file to the defined target directory inside the project's data folder
+
+        Args:
+            input_file (BinaryIO): the file
+            path (str): the target directory path
+        """
+        path = pathlib.Path(f"{self.files_directory_name}/{path}")
+        path.parent.mkdir(parents=True, exist_ok=True)
+        with open(path, mode="wb") as file:
+            input_file.seek(0)
+            file.write(input_file.read())
+
+    @staticmethod
+    def write_files(files: Dict[str, Any], target_directory: str) -> None:
+        """
+        Write files to a supplied target directory
+
+        Args:
+            files (Dict[str, Any]): a dict representing files needing to be written in the target directory
+                with key as the file url and the value as file content
+            target_directory (str): the directory all the files will reside in
+        """
+        for key, content in files.items():
+            path = pathlib.Path(f"{target_directory}/{key}")
+            path.parent.mkdir(parents=True, exist_ok=True)
+            with open(path, mode="wb") as file:
+                file.write(content)
+
+    @abstractmethod
+    def load_templates(self, target_directory: str, template_directory: str) -> None:
+        """
+        Args:
+            target_directory: Target directory to store the templates in
+            template_directory: Base directory
+        """
+        raise NotImplementedError
+
+
+class DiskFileStorage(PlatoFileStorage, ABC):
+    def __init__(self, data_directory: str):
+        super().__init__(data_directory)
+
+    def save_file(self, input_file: BinaryIO, path: str) -> None:
+        """
+        Write file into local storage folder
+
+        Args:
+            input_file (BinaryIO): the input file
+            path (str): the local storage path
+        """
+        self.write_file_locally(input_file, path)
+
+    def load_templates(self, target_directory: str, template_directory: str) -> None:
+        """
+            Args:
+                target_directory: Target directory to store the templates in
+                template_directory: Base directory
+        """
+        raise NotImplementedError
```

Comment:
Alternatively, we could have this with an empty body (I believe it's just "..." instead of "raise blabla", and we can remove the "if S3 -> load templates" on the main file. The storage class itself knows what it needs to do (nothing for local, stuff for S3)

Reply:
yes, pass will work here! I totally forgot about it